language_load.py

```diff
@@ -1,99 +1,152 @@
+    """ Updates an object translation, if applicable """
     file = open(filename, encoding="utf8")
     data = json.load(file)
     file.close()
+    obj_id = data['id']
 
-    if not 'strings' in data:
-        if verbose:
-            print("No strings in " + data['id'] + " -- skipping")
-        continue
-
-    if not data['id'] in strings_by_object:
-        if verbose:
-            print("No translations for " + data['id'] + " in dump file -- skipping")
-        continue
+    if not translatable(verbose, data):
+        return
+    if not is_object_translated(verbose, obj_id, strings_by_object):
+        return
 
     updated = False
     for string_key in data['strings']:
 
-        if not string_key in strings_by_object[data['id']]:
-            if verbose:
-                print("No translation for " + data['id'] + " string '" + string_key + "' in dump file -- skipping")
+        if not is_key_translated(verbose, obj_id, string_key, strings_by_object[obj_id]):
             continue
 
-        if not fallback_language in data['strings'][string_key]:
-            if verbose:
-                print("No en-GB reference for " + data['id'] + " string '" + string_key + "' in dump file -- probably shouldn't exist; skipping")
+        if not fallback_key_exists(verbose, ref_lang, string_key, obj_id,
+                                   data['strings'][string_key]):
             continue
 
-        if not language_to_import in data['strings'][string_key]:
-            if strings_by_object[data['id']][string_key] == data['strings'][string_key][fallback_language]:
-                # TODO: Is this desirable behaviour?
-                if verbose:
-                    print("Translation for " + data['id'] + " string '" + string_key + "' is identical to en-GB -- skipping")
-                continue
-
-            if verbose:
-                print("Adding " + data['id'] + " string '" + string_key + "'")
-            data['strings'][string_key][language_to_import] = strings_by_object[data['id']][string_key]
-            updated = True
-        else:
-            if strings_by_object[data['id']][string_key] == data['strings'][string_key][language_to_import]:
-                if verbose:
-                    print("Translation for " + data['id'] + " string '" + string_key + "' has not changed -- skipping")
-                continue
-
-            print("Updating " + data['id'] + " string '" + string_key + "'")
-            data['strings'][string_key][language_to_import] = strings_by_object[data['id']][string_key]
+        previously_translated = translation_existed(target_lang, data['strings'][string_key])
+        translation_needs_update = False
+        if previously_translated:
+            translation_needs_update = translation_changed(verbose, obj_id, string_key,\
+                                          data['strings'][string_key][target_lang],\
+                                          strings_by_object[obj_id][string_key])
```

Comment:
I appreciate the use of booleans and functions to improve the readability, however I think this could be improved. I'd prefer to not break expressions into several lines if we can avoid it. Perhaps something like:
```suggestion
            translation = data['strings'][string_key][target_lang]
            target_string = strings_by_object[obj_id][string_key]
            translation_needs_update = translation_changed(verbose, obj_id, string_key,\
                                          translation, target_string)
```